include/rocksdb/merge_operator.h

```diff
@@ -222,26 +222,38 @@ class MergeOperator {
   }
 };
 
-// The simpler, associative merge operator.
+// A merge operator that can be implemented as an associative binary
+// operator on values/operands. Implementations must satisfy
+//      merge_two_values(merge_two_values(a, b), c)
+//   == merge_two_values(a, merge_two_values(b, c))
+// as partial merges might substitute the latter for the former.
+//
```

Comment:
```
//      assoc_merge(assoc_merge(input_a, operand_b), operand_c)
//   == assoc_merge(input_a, assoc_merge(operand_b, operand_c))
// as an associative merge operator implements both PartialMerge and FullMerge in one function, it may be applied to either:
// 1) null (when there is no existing value) and an operand
// 2) an existing value and an operand
// 3) two operands
```